.github/workflows/build-kernel.yml

```diff
@@ -0,0 +1,43 @@
+name: Build Linux Kernel (siklu_arm64_defconfig)
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - tg-v6.12
+  pull_request:
+    branches:
+      - tg-v6.12
+
+jobs:
+  build:
```

Comment:
can you check if it is easy to cache the results? to reduce the build time? 
See https://github.com/actions/cache
you a can save the whole kernel build directory and and restore it.